### PR TITLE
create query operator and create query operator email test

### PR DESCRIPTION
**Pull request by ManhNTX**
create query operator and create query operator email test

### CODE REVIEW

File: lib/jmap/core/filter/filter_operator.dart [hunk 1]
```diff
@@ -2,7 +2,7 @@ import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
 
 abstract class FilterOperator extends Filter {
 
-  final Operator operator;
+  final String operator;
```
Comment (hoangdat) [inline lib/jmap/core/filter/filter_operator.dart:5]: IMO, you should still keep operator here, and create a `converter` for it.

File: lib/jmap/core/filter/operator/query_filter_operator.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'package:jmap_dart_client/jmap/core/filter/filter_operator.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'query_filter_operator.g.dart';
+
+@JsonSerializable(explicitToJson: true)
+class QueryFilterOperator extends FilterOperator {
```
Comment (hoangdat) [inline lib/jmap/core/filter/operator/query_filter_operator.dart:7]: `LogicFilterOperator`

File: lib/http/converter/filter_conveter.dart [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  const FilterConverter();
+
+  @override
+  Filter fromJson(Map<String, dynamic> json) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Map<String, dynamic> toJson(Filter object) {
+    return object.toJson();
```
Comment (hoangdat) [inline lib/http/converter/filter_conveter.dart:14]: 👍

File: lib/http/converter/filter_conveter.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+class FilterConverter implements JsonConverter<Filter, Map<String, dynamic>> {
+  const FilterConverter();
+
+  @override
+  Filter fromJson(Map<String, dynamic> json) {
+    throw UnimplementedError();
```
Comment (hoangdat) [inline lib/http/converter/filter_conveter.dart:9]: any risk for us?

**Review by hoangdat — APPROVED** [review]

File: lib/jmap/core/filter/operator/logic_filter_operator.dart [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter_operator.dart';
+
+class LogicFilterOperator extends FilterOperator {
+  LogicFilterOperator(Operator operator,Set<Filter> conditions)
```
Comment (dab246) [inline lib/jmap/core/filter/operator/logic_filter_operator.dart:5]: There should be a space between the second parameter and the comma before it.
`LogicFilterOperator(Operator operator, Set<Filter> conditions)`

File: lib/jmap/core/filter/operator/logic_filter_operator.dart [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+      : super(operator, conditions);
+
+  @override
+  List<Object?> get props => [
+        operator,
```
Comment (dab246) [inline lib/jmap/core/filter/operator/logic_filter_operator.dart:10]: Indent a bit too much

File: lib/jmap/core/filter/operator/logic_filter_operator.dart [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        conditions,
+      ];
+
+  @override
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'operator': operator.name,
```
Comment (dab246) [inline lib/jmap/core/filter/operator/logic_filter_operator.dart:16]: same

**Review by dab246 — APPROVED** [review]
/lgtm

**Pull request merged**